web/Services/CsvImportService.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System.Data;
+using BricksAndHearts.Database;
+using BricksAndHearts.ViewModels;
+using LINQtoCSV;
+using Microsoft.EntityFrameworkCore;
+
+namespace BricksAndHearts.Services;
+
+public interface ICsvImportService
+{
+    public (List<string> FlashTypes, List<string> FlashMessages) CheckIfImportWorks(IFormFile csvFile);
+    public Task<(List<string> FlashTypes, List<string> FlashMessages)> ImportTenants(IFormFile csvFile, (List<string> flashTypes, List<string> flashMessages) flashResponse);
+}
+
+public class CsvImportService : ICsvImportService
+{
+    private readonly BricksAndHeartsDbContext _dbContext;
+    private readonly ILogger<CsvImportService> _logger;
+
+    public CsvImportService(BricksAndHeartsDbContext dbContext, ILogger<CsvImportService> logger)
+    {
+        _logger = logger;
+        _dbContext = dbContext;
+    }
+
+    public (List<string>, List<string>) CheckIfImportWorks(IFormFile csvFile)
+    {
+        List<string> flashTypes = new(),
+            flashMessages = new();
+        var headerLine = "";
+        using (var streamReader = new StreamReader(csvFile.OpenReadStream()))
+        {
+            headerLine = streamReader.ReadLine();
+        }
+        
+        var headers = headerLine!.Split(",");
+        var databaseHeaders = new List<string>();
+        
+        //check for missing columns in csv file
+        foreach (var dataProp in typeof(TenantUploadModel).GetProperties())
+        {
+            if (!headers.Contains(dataProp.Name) && dataProp.Name != "Id")
+            {
+                _logger.LogWarning($"Missing column: {dataProp.Name}");
+                flashTypes.Add("danger");
+                flashMessages.Add(
+                    $"Import has failed because column {dataProp.Name} is missing. Please add this column to your records before attempting to import them.");
+            }
+            databaseHeaders.Add(dataProp.Name);
+        }
+
+        //check for extra columns in csv file
+        foreach (var header in headers)
+        {
+            if (!databaseHeaders.Contains(header))
+            {
+                _logger.LogWarning($"Extra column: {header}");
+                flashTypes.Add("warning");
+                flashMessages.Add(
+                    $"The column {header} does not exist in the database. All data in this column has been ignored.");

```

Comment:
Would be slightly clearer, especially if the problem is trailing whitespace or something.
```suggestion
                    $"The column \"{header}\" does not exist in the database. All data in this column has been ignored.");
```